addons/project_mrp/models/project_project.py

```diff
@@ -1,101 +1,70 @@
-# -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models, _lt
-from odoo.osv import expression
+from odoo import _, fields, models
 
 
-class Project(models.Model):
-    _inherit = "project.project"
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
 
-    production_count = fields.Integer(related="analytic_account_id.production_count", groups='mrp.group_mrp_user', export_string_translation=False)
-    workorder_count = fields.Integer(related="analytic_account_id.workorder_count", groups='mrp.group_mrp_user', export_string_translation=False)
-    bom_count = fields.Integer(related="analytic_account_id.bom_count", groups='mrp.group_mrp_user', export_string_translation=False)
+    bom_count = fields.Integer(compute='_compute_bom_count', groups='mrp.group_mrp_user', export_string_translation=False)
+    production_count = fields.Integer(compute='_compute_production_count', groups='mrp.group_mrp_user', export_string_translation=False)
 
-    def action_view_mrp_production(self):
-        self.ensure_one()
-        action = self.env['ir.actions.actions']._for_xml_id('mrp.mrp_production_action')
-        action['domain'] = [('id', 'in', self.analytic_account_id.production_ids.ids)]
-        action['context'] = {'default_analytic_account_id': self.analytic_account_id.id}
-        if self.production_count == 1:
-            action['view_mode'] = 'form'
-            action['res_id'] = self.analytic_account_id.production_ids.id
-            if 'views' in action:
-                action['views'] = [
-                    (view_id, view_type)
-                    for view_id, view_type in action['views']
-                    if view_type == 'form'
-                ] or [False, 'form']
-        return action
+    def _compute_bom_count(self):
+        bomn_count_per_project = {
+            project.id: count
+            for project, count in self.env['mrp.bom']._read_group(
+                [('project_id', 'in', self.ids)],
+                ['project_id'], ['__count']
+            )
+        }
+        for project in self:
+            project.bom_count = bomn_count_per_project.get(project.id)
```

Comment:
Normally, we could have a recordset as a key. 🙂 

(Also, there is a typo on `bomn_count_per_project` variable name 😇 🤓 )
```suggestion
        bom_count_per_project = dict(
            self.env['mrp.bom']._read_group(
                [('project_id', 'in', self.ids)],
                ['project_id'], ['__count']
            )
        )
        for project in self:
            project.bom_count = bom_count_per_project.get(project)
```

Ah maybe it does not work because of the newId? 🧐

Reply:
No I think your suggestion's right